user/src/components/ApplicationForms/Stage/Stage.tsx

```diff
@@ -0,0 +1,277 @@
+
+  const filteredSunny1 = useFilteredStageOptions({
+    allStages: sunnyStageOptions,
+    selectedId: sunnySecondChoice,
+  });
+
+  const filteredSunny2 = useFilteredStageOptions({
+    allStages: sunnyStageOptions,
+    selectedId: sunnyFirstChoice,
+  });
+
+  const filteredRainy1 = useFilteredStageOptions({
+    allStages: rainyStageOptions,
+    selectedId: rainySecondChoice,
+  });
+
+  const filteredRainy2 = useFilteredStageOptions({
+    allStages: rainyStageOptions,
+    selectedId: rainyFirstChoice,
+  });
+
+  // エラー表示
+  const renderError = (error?: string) => {
+    return error ? <p className="text-[#FF0000] text-xs">{error}</p> : null;
+  };
+
+  // 一時保存処理
+  const handleSaveDraft = () => {
+    if (saveDraft()) {
+      alert('下書きを保存しました');
+    }
+  };
+
+  // 登録処理
+  const handleRegister = async () => {
+    if (!isValid) return;
+    setSubmitError('');
+
+    try {
+      const baseOrderData = {
+        group_id: currentGroupId,
+        fes_date_id: parseInt(date),
+        use_time_interval: performTime,
+        prepare_time_interval: prepTime,
+        cleanup_time_interval: cleanupTime,
+        remarks: remarks,
+      };
+
+      const sunnyOrderData: StageOrderData = {
+        ...baseOrderData,
+        is_sunny: true,
+        stage_first: parseInt(sunnyFirstChoice),
+        stage_second: parseInt(sunnySecondChoice),
+      };
+
+      const rainyOrderData: StageOrderData = {
+        ...baseOrderData,
+        is_sunny: false,
+        stage_first: parseInt(rainyFirstChoice),
+        stage_second: parseInt(rainySecondChoice),
+      };
+
+      const result = await submitStageOrder(sunnyOrderData, rainyOrderData);
+
+      if (result.success) {
+        alert('ステージ希望を登録しました。');
+        clearDraft();
+      } else {
+        setSubmitError(
+          '送信中にエラーが発生しました。もう一度お試しください。'
+        );
+      }
+    } catch (error) {
+      setSubmitError('予期せぬエラーが発生しました。もう一度お試しください。');
+    }
+  };
+
+  return (
+    <div className="w-fit flex flex-col rounded-[20px] gap-10 p-20 text-black bg-white shadow-[0px_4px_6px_0px_rgba(0,_0,_0,_0.25)]">
+      {hasError && (
+        <div className="w-fit bg-red-100 border border-red-400 text-red-700 px-4 py-3 rounded relative">
+          <strong className="font-bold">エラー：</strong>
+          <span className="block sm:inline">
+            データの取得に失敗しました。ページを再読込してください。
+          </span>
+        </div>
+      )}
+
+      {submitError && (
+        <div className="w-fit bg-red-100 border border-red-400 text-red-700 px-4 py-3 rounded relative">
+          <strong className="font-bold">送信エラー：</strong>
+          <span className="block sm:inline">{submitError}</span>
+        </div>
+      )}
+
+      {isLoading && (
+        <div className="text-center py-4">
+          <p>データを読み込み中です...</p>
+        </div>
+      )}
+
+      <div>
+        <Radio
+          label="開催日"
+          value={date}
+          onChange={(value) => updateField('date', value)}
+          required={true}
```

Comment:
```suggestion
          required
```
これだけでtrueに判定されるはず
他の場所もお願いします！